src/Widgets/ContractorButtonWithMenu.vala

```diff
@@ -79,23 +79,22 @@ namespace Maya.View.Widgets {
         private void savecal () {
             /* creates a .ics file */
             Util.save_temp_selected_calendars ();
-            filechooser = new Gtk.FileChooserDialog (_("Export Calendar…"), null, Gtk.FileChooserAction.SAVE);
-            var filter = new Gtk.FileFilter ();
-            filter.add_mime_type("text/calendar");
-            filechooser.set_current_name(_("calendar.ics"));
-            filechooser.set_filter (filter);
-            filechooser.add_button (_("Cancel"), Gtk.ResponseType.CLOSE);
-            filechooser.add_button (_("Save"), Gtk.ResponseType.APPLY);
-            filechooser.response.connect (on_response);
-            filechooser.show_all ();
-            filechooser.run ();
-
 
-        }
+            var filter = new Gtk.FileFilter ();
+            filter.add_mime_type ("text/calendar");
+
+            filechooser = new Gtk.FileChooserNative (
+                _("Export Calendar…"),
+                null,
+                Gtk.FileChooserAction.SAVE,
+                _("Save"),
+                _("Cancel")
+            );
+            filechooser.do_overwrite_confirmation = true;
```

Comment:
Not supported across portal ?? (to be confirmed).

Reply:
Yeah, the new native chooser should probably just do this and not provide an option. This is only for the fallback chooser